apps/dashboard/src/routes/courses/[id]/lessons/[...lessonParams]/+page.svelte

```diff
@@ -240,6 +246,50 @@
     $apps.open = true;
   }
 
+  const goToNextLesson = () => {
+    const currentLessonIdx = $lessons.findIndex((lesson) => lesson.id === data.lessonId);
+
+    if (currentLessonIdx !== -1 && currentLessonIdx + 1 < $lessons.length) {
+      const nextLesson = $lessons[currentLessonIdx + 1];
+      console.log('next lesson', nextLesson);
+
+      fetchReqData(nextLesson.id, true);
+      const nextLessonUrl =
+        $globalStore.isStudent && !nextLesson.is_unlocked
+          ? $page.url.pathname
+          : `/courses/${$course.id}/lessons/${nextLesson.id}`;
+
+      goto(nextLessonUrl);
+    }
+  };
+
+  const goToPrevLesson = () => {
+    const currentLessonIdx = $lessons.findIndex((lesson) => lesson.id === data.lessonId);
+
+    if (currentLessonIdx !== -1 && currentLessonIdx - 1 >= 0) {
+      const prevLesson = $lessons[currentLessonIdx - 1];
+      console.log('next lesson', prevLesson);
+
+      fetchReqData(prevLesson.id, true);
+      const prevLessonUrl =
```

Comment:
All these logic as so complex. and you have a ton of repeated code.

The logic below is built upon the previous comment I made about the reactivity.

```ts
const goToNextOrPrevLesson = (lessonId: string, isPrev: boolean) => {
  const isDisabled = isNextOrPrevDisabled(lessonId, isPrev);

  // Always use early return
  if (isDisabled) return;
  
  const index = $lessons.findIndex((lesson) => lesson.id === lessonId);
  const nextOrPrevLesson = isPrev ? $lessons[index - 1] : $lessons[index + 1];

  const isLocked = $globalStore.isStudent && !nextOrPrevLesson.is_unlocked;
  // Early return for locked lesson
  if (isLocked) return;

  const path = `/courses/${$course.id}/lessons/${nextOrPrevLesson.id}`;
  // fetchReqData(nextLesson.id, true); - I don't know why you need this though
  goto(path);
}
```